docs/source/fetching/persisted-queries.mdx

```diff
@@ -0,0 +1,160 @@
+---
+title: Persisted Queries
+description: Secure your graph while minimizing request latency
+---
+
+Apollo supports two separate but related features called **persisted queries** and **automatic persisted queries** (APQs).
+With both features, clients can execute a GraphQL operation by sending an operation's ID instead of the entire operation string. An operation's ID is the SHA256 hash of the full string. Querying by ID can significantly reduce latency and bandwidth usage for very large operation strings.
+
+## Differences between persisted queries and APQ
+
+The persisted queries feature requires operations to be preregistered in a **persisted query list** (**PQL**). 
+This allows the PQL to act as a safelist of trusted operations made by your first-party apps. As such, persisted queries is a security feature as much as a performance one.
+
+With APQs, if the server can't find the operation ID the client provides, the server returns an error indicating that it needs the full operation string. If the Apollo iOS client receives this error, it automatically retries the operation with the full operation string.
+
+If you _only_ want to improve request latency and bandwidth usage, APQ addresses your use case. If you _also_ want to secure your supergraph with operation safelisting, you should preregister trusted operations in a PQL.
+
+For more details on differences between persisted queries and APQ, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries#differences-from-automatic-persisted).
+
+### Requirements
+
+Persisted queries is currently in [preview](/resources/product-launch-stages#preview) and has the following requirements:
+- Apollo iOS (v1.3.1+)
+- [Apollo Router](/router) (v1.25.0+)
+- [GraphOS Enterprise plan](/graphos/enterprise/)
+
+You can use APQ with the following versions of Apollo iOS, Apollo Server, and Apollo Router:
+- Apollo iOS (v1.0.0+)
+- [Apollo Server](/apollo-server/) (v1.0.0+)
+- [Apollo Router](/router) (v0.1.0+)
+
+> **Note:** You can use _either_ Apollo Server _or_ Apollo Router for APQs. They don't need to be used together.
+
+## Implementation steps
+
+Both persisted queries and APQs require you to configure code generation and how your client makes requests. If you intend to use persisted queries for safelisting, you also need to generate an operations manifest.
+
+We recommend you follow this order while implementing: 
+
+1. **For persisted queries only**: Generate and publish operation manifests
+2. **For both persisted queries and APQs**: Configure generated operation models
+3. **For both persisted queries and APQs**: Enable persisted queries on `ApolloClient`
```

Comment:
There is no reason generating and publishing the manifest needs to happen first. And I think it would be clearer that it's only necessary for persisted queries if we put it last like this:

```suggestion
To implement these features, follow these steps.

1. Configure generated operation models
2. Enable persisted queries on `ApolloClient`
3. **For persisted queries only**: Generate and publish operation manifests
```

And then move the "Generate Operation Manifests" section to the bottom.

Reply:
I'm recognizing that this has been added as one of the pieces that we are going to replace with shared components . So if this suggested change is preferred by others, we should make the change there and then use that component.

Reply:
I think this order makes the most sense,  even if it sticks the PQ-only step in the middle.
1 - Code generation
2 - Publish operation manifest
3 - Client updates 

(It also matches the implementation narrative in the "[main](https://github.com/apollographql/docs/pull/496/files)" PQ docs, though I'm open to changing it if it's not the most user-intuitive.)

If the sole purpose of putting manifest publication last is to make clearer which step is PQ only, I'd say it'd be better to find another way to make this clearer.